### PR TITLE
[Dialogs] Publish Customize API

### LINKED ISSUE #9869: [Dialogs] Internal issue: b/150085275
This was filed as an internal issue. If you are a Googler, please visit [b/150085275](http://b/150085275) for more details.

---
#### Internal data
- Associated internal bug: [b/150085275](http://b/150085275)
- Blocked by: https://github.com/material-components/material-components-ios/issues/9890

### PR DESCRIPTION
**Pull request by galiak11**
# Description
Publishing Dialogs' `titleIconView` API.

API in the MDCAlertController+Customize.h are not included in the Dialogs umbrella headers. and must be implicitly imported when used.

## Issues
Fixes: #9869
b/150085275

### CODE REVIEW

File: components/Dialogs/src/MDCAlertController+Customize.h [hunk 1]
```diff
@@ -14,14 +14,19 @@
 
 #import "MaterialDialogs.h"
 
+/**
+ The Customize extensions privde APIs that extend the default Material Dialog. Make sure to adjust
```
Comment (jverkoey) [inline components/Dialogs/src/MDCAlertController+Customize.h:18]: provide*

File: components/Dialogs/src/MDCAlertController+Customize.h [hunk 2]
```diff
@@ -14,14 +14,19 @@
...
+ your custom theming when using these APIs, if you are concerned with conforming to the Material
```
Comment (jverkoey) [inline components/Dialogs/src/MDCAlertController+Customize.h:19]: What does "Make sure to adjust your custom theming" mean in this context? If we're not able to explain in this context, can we link the reader to some additional documentation?

File: components/Dialogs/src/MDCAlertController+Customize.h [hunk 3]
```diff
@@ -14,14 +14,19 @@
...
+ spec (material.io).
+*/
 @interface MDCAlertController (Customize)
 
 /**
- An optional custom icon view above the title of the alert.
+ An optional custom icon view that's displayed above the title of the alert, enabling custom
+ implementations of the view above the title.
```
Comment (jverkoey) [inline components/Dialogs/src/MDCAlertController+Customize.h:26]: The ", enabling custom implementations of the view above the title." part is implied and can be removed.

File: components/Dialogs/src/MDCAlertController+Customize.h [hunk 4]
```diff
@@ -14,14 +14,19 @@
...
 
- @note This property is intended to be used to provide a custom implementation of the title icon
- view. If the intention is to just display a `UIImage`, use `setTitleIcon:` API instead. If
- 'titleIcon' is set, 'titleIconView' is ignored.
+ @note Use `titleIcon` to display icons or images.  Use `titleIconView` for custom views
+ implementations. If both `titleIcon` are `titleIconView` are set, 'titleIconView' is ignored.
```
Comment (jverkoey) [inline components/Dialogs/src/MDCAlertController+Customize.h:29]: > If both `titleIcon` are `titleIconView` are set, 'titleIconView' is ignored.

This behavior is surprising. E.g. with navigationItem, if you provide both a title and a titleView, titleView takes precedent (https://developer.apple.com/documentation/uikit/uinavigationitem/1624935-titleview?language=objc). Please align with UIKit's behavior in this case and make titleIconView take precedence in a PR prior to landing this one.

**Review by jverkoey — CHANGES_REQUESTED** [review]

**Comment by galiak11** [comment]
Closing because this was implemented in "cl/303284455 - [Dialogs] Publish Customize API".

**Pull request closed without merging**